File: app/src/main/java/xyz/miaoguoge/musicplayer/Global.java
package xyz.miaoguoge.musicplayer;

import android.app.Application;

import java.util.ArrayList;
import java.util.List;

public class Global extends Application {
    public static List<Song> Favor = new ArrayList<>();
    public static List<Song> Recent= new ArrayList<>();
    public static List<Song> All= new ArrayList<>();
}

File: app/src/main/java/xyz/miaoguoge/musicplayer/Global.kt
package xyz.miaoguoge.musicplayer

import android.app.Application
import xyz.miaoguoge.musicplayer.Song
import java.util.ArrayList

object Global : Application() {
    var Favor: ArrayList<Song> = ArrayList()
    var Recent: ArrayList<Song> = ArrayList()
    var All: ArrayList<Song> = ArrayList()

    fun getSongByFilename(filename: String): Song? {
        for (song in All) {
            if (song.Filename == filename) {
                return song
            }
        }
        return null
    }
}
